src/main/java/com/openshift/jenkins/plugins/pipeline/model/RetryIClient.java

```diff
@@ -0,0 +1,241 @@
+package com.openshift.jenkins.plugins.pipeline.model;
+
+import hudson.model.TaskListener;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import com.openshift.jenkins.plugins.pipeline.MessageConstants;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.IOpenShiftWatchListener;
+import com.openshift.restclient.IResourceFactory;
+import com.openshift.restclient.IWatcher;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.UnsupportedVersionException;
+import com.openshift.restclient.api.ITypeFactory;
+import com.openshift.restclient.authorization.IAuthorizationContext;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.ICapability;
+import com.openshift.restclient.model.IList;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.JSONSerializeable;
+
+public class RetryIClient implements IClient {
+    
+    private static int MAX_RETRY = 3;
+
+    private final IClient client;
+    private final TaskListener listener;
+    
+    private Object retry(Callable<Object> call) throws OpenShiftException {
+        int retryCount = 0;
+        OpenShiftException ose = null;
+        while (retryCount < MAX_RETRY) {
+            try {
+                Object o = call.call();
+                return o;
```

Comment:
I assume standard failures like "permission denied" or "unknown resource" do not result in exceptions from call()?